tests/nnstreamer_filter_extensions_common/unittest_tizen_template.cc

```diff
@@ -149,30 +149,36 @@ TEST (nnstreamer_EXT_NICK_NAME_basic_functions, get_dimension_fail_n)
   GstTensorFilterProperties prop = {
     .fwname = "EXT_NAME",
     .fw_opened = 0,
-    .model_files = const_cast<const char **>(model_files),
+    .model_files = const_cast < const char **>(model_files),
     .num_models = (int) g_strv_length (model_files),
   };
 
   const GstTensorFilterFramework *sp = nnstreamer_filter_find ("EXT_NAME");
   EXPECT_NE (sp, (void *) NULL);
 
-  /** get input/output dimension without open */
-  ret = sp->getInputDimension (&prop, &data, &res);
-  EXPECT_NE (ret, 0);
-  ret = sp->getOutputDimension (&prop, &data, &res);
-  EXPECT_NE (ret, 0);
+  if (sp->getInputDimension && sp->getOutputDimension) {
+    /** get input/output dimension without open */
+    ret = sp->getInputDimension (&prop, &data, &res);
+    EXPECT_NE (ret, 0);
+    ret = sp->getOutputDimension (&prop, &data, &res);
+    EXPECT_NE (ret, 0);
 
-  ret = sp->open (&prop, &data);
-  EXPECT_EQ (ret, 0);
+    ret = sp->open (&prop, &data);
+    EXPECT_EQ (ret, 0);
 
-  /** get input/output dimension unsuccessfully */
-  ret = sp->getInputDimension (&prop, &data, NULL);
-  EXPECT_NE (ret, 0);
+    /** python subplugin modifies methods based on model file */
+    if (sp->getInputDimension && sp->getOutputDimension) {
```

Comment:
unnecessary condition (duplicated with line 159)

Reply:
In the current implementation, extensions can change their functions based on the model files (this behavior is raised in #2034). Until this behavior is changed, this condition is needed.

Reply:
I understand